sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1116,51 +1110,12 @@ public Write withMaxOutstandingBytes(long bytes) {
      * always enabled on batch writes and limits the number of outstanding requests to the Bigtable
      * server.
      *
-     * <p>When enabled, will also set default {@link #withThrottlingReportTargetMs} to 1 minute.
-     * This enables runner react with increased latency in flush call due to flow control.
-     *
      * <p>Does not modify this object.
      */
     public Write withFlowControl(boolean enableFlowControl) {
-      BigtableWriteOptions options = getBigtableWriteOptions();
-      BigtableWriteOptions.Builder builder = options.toBuilder().setFlowControl(enableFlowControl);
-      if (enableFlowControl) {
-        builder = builder.setThrottlingReportTargetMs(60_000);
-      }
-      return toBuilder().setBigtableWriteOptions(builder.build()).build();
-    }
-
-    /**
-     * Returns a new {@link BigtableIO.Write} with client side latency based throttling enabled.
-     *
-     * <p>Will also set {@link #withThrottlingReportTargetMs} to the same value.
-     */
-    public Write withThrottlingTargetMs(int throttlingTargetMs) {
```

Comment:
can you actually remove a public method? I suspect that you need to make it a no-op that logs a warning

Reply:
I think I can and we should. 

Only user actively set this method in their 2.59.0 pipeline will be affected. Users do not explicitly set this method, and upgrade from <=2.58.0 are unaffected. For those indeed use this method in 2.59.0 they have good reason to do so. We are removing a tested and functioning feature on request, make it no-op or warning then user won't aware this feature is gone in the next release.

In the past Beam has an "Experimental" annotation but voted to remove them (#26490). Now "new code is changeable/evolving by default" (see discussion link in that PR) especially true for a new API in single version not enabled by default.

Reply:
If the conceptual "binary compatibility" is a concern, I've add back the method, but throw instead of no-op or ignore there

Reply:
yes, I was concerned about breaking binary compatibility. I 

I dont see it throwing an exception though, nor do I think it should throw an exception. Please make it a no-op and log a warning instead.
Also we didnt expose ThrottlingTargetMs nor latency based throttling for a reason, it has some really sharp edge cases that are very hard to debug. Please make this a no-op as it was prior to your change